internal/cephfs/core/filesystem.go

```diff
@@ -25,76 +25,102 @@ import (
 	"github.com/ceph/ceph-csi/internal/util/log"
 )
 
-func (vo *VolumeOptions) getFscID(ctx context.Context) (int64, error) {
-	fsa, err := vo.conn.GetFSAdmin()
+// FileSystem is the interface that holds the signature of filesystem methods
+// that interacts with CephFS filesystem API's.
+type FileSystem interface {
+	// GetFscID returns the ID of the filesystem with the given name.
+	GetFscID(context.Context, string) (int64, error)
+	// GetMetadataPool returns the metadata pool name of the filesystem with the given name.
+	GetMetadataPool(context.Context, string) (string, error)
+	// GetFsName returns the name of the filesystem with the given ID.
+	GetFsName(context.Context, int64) (string, error)
+}
+
+// fileSystem is the implementation of FileSystem interface.
+type fileSystem struct {
```

Comment:
Thinking about this a little more, could the `VolumeOptions` type not just implement this interface? I think the `VolumeOptions` struct is misnamed anyway, it probably should be a struct called `CephFSVolume` or something...

Reply:
the final idea is to have independent struck like  Volume, Snapshot, etc, and move the duplicate or just use the core structs in VolumeOptions how we do for rbdImage and rbdVolume. as there is no dependency with VolumeOptions and Filesystem just making use of empty struct here.